### PR TITLE
Fix orderbook notification while disconnected

### PR DESCRIPTION
**Pull request by wltsmrz**
OrderBook's `notify()` function is the entry point for orderbook notifications (created, modified, deleted offers). In this function,  `_subscribed` property is checked to be truthy, however, this property only means that the end user subscribed or unsubscribed. The orderbook can also disconnect automatically, if a connection to rippled is lost. When this happens, all of the orderbook's account funds caches are cleared, but `notify()` would still execute as normal, because the end user hasn't unsubscribed. So, with cleared caches, the default return value for cached account funds is `0`. But if `notify()` detects a deleted offer... you get it. The error is `Offer total cannot be negative` https://github.com/ripple/ripple-lib/issues/352

This pull request simply adds a check for `_synced` property, which is set to `false` on disconnect

### CODE REVIEW

File: src/core/orderbook.js [hunk 1]
```diff
@@ -842,7 +842,7 @@ OrderBook.prototype.updateOwnerOffersFundedAmount = function(account) {
 OrderBook.prototype.notify = function(transaction) {
   const self = this;
 
-  if (!this._subscribed) {
+  if (!(this._subscribed && this._synced)) {
```
Comment (clark800) [inline src/core/orderbook.js]: I was just looking at this issue too. It was possible for a `notify` to create a `model` event before it gets synchronized.

**Comment by clark800** [comment]
Can you summarize what the problem was and what the fix was? It's not obvious from the changes.

**Comment by wltsmrz** [comment]
I think Nicholas Dudfield nailed it earlier, but I'll summarize in the description of this pull request

**Comment by clark800** [comment]
LGTM
